pkg/aws/instance_info.go

```diff
@@ -124,34 +124,34 @@ func (types *InstanceTypes) SyntheticInstanceInfo(instanceTypes []string) (Insta
 		return Instance{}, fmt.Errorf("no instance types provided")
 	} else if len(instanceTypes) == 1 {
 		return types.InstanceInfo(instanceTypes[0])
-	} else {
-		first, err := types.InstanceInfo(instanceTypes[0])
+	}
+
+	first, err := types.InstanceInfo(instanceTypes[0])
+	if err != nil {
+		return Instance{}, err
+	}
+
+	result := Instance{
+		InstanceType:              first.InstanceType,
+		VCPU:                      first.VCPU,
+		Memory:                    first.Memory,
+		InstanceStorageDevices:    first.InstanceStorageDevices,
+		InstanceStorageDeviceSize: first.InstanceStorageDeviceSize,
+		Architecture:              first.Architecture,
+	}
+	for _, instanceType := range instanceTypes[1:] {
+		info, err := types.InstanceInfo(instanceType)
 		if err != nil {
 			return Instance{}, err
 		}
 
-		result := Instance{
-			InstanceType:              first.InstanceType,
-			VCPU:                      first.VCPU,
-			Memory:                    first.Memory,
-			InstanceStorageDevices:    first.InstanceStorageDevices,
-			InstanceStorageDeviceSize: first.InstanceStorageDeviceSize,
-			Architecture:              first.Architecture,
-		}
-		for _, instanceType := range instanceTypes[1:] {
-			info, err := types.InstanceInfo(instanceType)
-			if err != nil {
-				return Instance{}, err
-			}
-
-			result.VCPU = min(result.VCPU, info.VCPU)
-			result.Memory = min(result.Memory, info.Memory)
-			result.InstanceStorageDeviceSize = min(result.InstanceStorageDeviceSize, info.InstanceStorageDeviceSize)
-			result.InstanceStorageDevices = min(result.InstanceStorageDevices, info.InstanceStorageDevices)
-		}
-		result.InstanceType = "<multiple>"
-		return result, nil
+		result.VCPU = min(result.VCPU, info.VCPU)
+		result.Memory = min(result.Memory, info.Memory)
+		result.InstanceStorageDeviceSize = min(result.InstanceStorageDeviceSize, info.InstanceStorageDeviceSize)
+		result.InstanceStorageDevices = min(result.InstanceStorageDevices, info.InstanceStorageDevices)
 	}
+	result.InstanceType = "<multiple>"
```

Comment:
So if there is only one instance type, we have now `<multiple>`

Reply:
ah no in line 125 we check that we have one and return early. This means you can move result.InstanceType to the top.

Reply:
There is a check `if len(instanceTypes) == 1` above. The change is equivalent as it just removes else branch where other if branches have return

Reply:
Fixed